From 6a3961779f1d0af8effbe08664664437478aaa5d Mon Sep 17 00:00:00 2001
From: Freddie Rice <freddie.rice@gmail.com>
Date: Fri, 19 Oct 2018 08:02:12 +0200
Subject: [PATCH] anaconda: xen.efi upgraded during each install
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Frédéric Pierret (fepitre) <frederic.epitre@orange.fr>
---
 pyanaconda/bootloader.py | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/pyanaconda/bootloader.py b/pyanaconda/bootloader.py
index a07237dcc..f3627e7dc 100644
--- a/pyanaconda/bootloader.py
+++ b/pyanaconda/bootloader.py
@@ -1938,12 +1938,15 @@ class XenEFI(EFIGRUB):
             boot_part_num = self.stage1_device.parents[0].parted_partition.number
         boot_part_num = str(boot_part_num)
 
-        if not os.path.exists(
-                "{}/{}".format(util.getSysroot() + self.config_dir, "xen.efi")):
-            xen_efi = [x for x in os.listdir(util.getSysroot() + self.config_dir) if
-                       x.startswith('xen-') and x.endswith('.efi')][0]
-            shutil.copy("{}/{}".format(util.getSysroot() + self.config_dir, xen_efi),
-                        "{}/{}".format(util.getSysroot() + self.config_dir, "xen.efi"))
+        # could be an old version, replace in case
+        xen_efi_target = "{}/{}".format(util.getSysroot() + self.config_dir, "xen.efi")
+        if os.path.exists(xen_efi_target):
+            os.remove(xen_efi_target)
+
+        xen_efi = [x for x in os.listdir(util.getSysroot() + self.config_dir) if
+                   x.startswith('xen-') and x.endswith('.efi')][0]
+        shutil.copy("{}/{}".format(util.getSysroot() + self.config_dir, xen_efi),
+                    "{}/{}".format(util.getSysroot() + self.config_dir, "xen.efi"))
         rc = self.efibootmgr("-c", "-w", "-L", productName,
                              "-d", boot_disk.path, "-p", boot_part_num,
                              "-l",
-- 
2.19.2

